metrics/src/lib.rs

```diff
@@ -99,7 +99,13 @@ pub fn initialize() -> Option<tokio::task::JoinHandle<()>> {
 
 #[cfg(not(feature = "prometheus"))]
 pub fn initialize() -> Option<tokio::task::JoinHandle<()>> {
+    #[cfg(not(feature = "test"))]
     metrics::set_recorder(&crate::stats::NODE_STATS).expect("couldn't initialize the metrics recorder!");
 
+    // Let the errors through in case the recorder has already been set as is likely when running
```

Comment:
does this still apply? I thought the single-threaded mode for these tests was there to start with a clean slate every time

Reply:
It does since the recorder is global (and leaks) even when running single threaded. Since we need to initialize the metrics for each test (as we don't want to depend on the running order of the tests), this call will error for all cases except the first in the run.